diff --git a/iMage/jmjrst.main/src/main/java/org/jis/generator/Generator.java b/iMage/jmjrst.main/src/main/java/org/jis/generator/Generator.java
index 87a83a1..339ecab 100644
--- a/iMage/jmjrst.main/src/main/java/org/jis/generator/Generator.java
+++ b/iMage/jmjrst.main/src/main/java/org/jis/generator/Generator.java
@@ -68,6 +68,10 @@ import org.jis.options.Options;
 public class Generator {
   public final static double ROTATE_90  = Math.toRadians(90);
   public final static double ROTATE_270 = Math.toRadians(270);
+  
+  //added a variable for -90 and -270 degrees
+  public final static double ROTATE_Minus90 = Math.toRadians(-90);
+  public final static double ROTATE_Minus270 = Math.toRadians(-270);
 
   private Main               m;
   private Options            o;
@@ -740,6 +744,22 @@ public class Generator {
       transform.rotate(Generator.ROTATE_270);
       width = image.getHeight(null); // swap
       height = image.getWidth(null);
+      
+    } else if (rotate == ROTATE_Minus90){
+    	
+    	//Added: Case of rotating the image -90 degrees.
+    	transform.translate(0, width);
+    	transform.rotate(Generator.ROTATE_Minus90);
+    	width = image.getHeight(null); // swap
+        height = image.getWidth(null);
+        
+    } else if (rotate == ROTATE_Minus270) {
+    	
+    	transform.translate(height, 0);
+        transform.rotate(Generator.ROTATE_Minus270);
+        width = image.getHeight(); // swap
+        height = image.getWidth();
+    	
     }
     else
     {
